python/lsst/meas/base/baseMeasurement.py

```diff
@@ -0,0 +1,258 @@
+
+        @param[in]  algMetadata     An lsst.daf.base.PropertyList that will be filled with metadata
+                                    about the plugins being run.  If None, an empty PropertyList will
+                                    be created.
+        @param[in]  **kwds          Additional arguments passed to lsst.pipe.base.Task.__init__.
+
+        This attaches two public attributes to the class for use by derived classes and parent tasks:
+         - plugins: an empty PluginMap, which will eventually contain all active plugins that will by
+           invoked by the run() method (to be filled by subclasses).  This should be considered read-only.
+         - algMetadata: a lsst.daf.base.PropertyList that will contain additional information about the
+           active plugins to be saved with the output catalog (to be filled by subclasses).
+        """
+        super(BaseMeasurementTask, self).__init__(**kwds)
+        self.plugins = PluginMap()
+        if algMetadata is None:
+            algMetadata = lsst.daf.base.PropertyList()
+        self.algMetadata = algMetadata
+
+    def initializePlugins(self, **kwds):
+        """Initialize the plugins (and slots) according to the configuration.
+
+        Derived class constructors should call this method to fill the self.plugins
+        attribute and add correspond output fields and slot aliases to the output schema.
+
+        In addition to the attributes added by BaseMeasurementTask.__init__, a self.schema
+        attribute holding the output schema must also be present before this method is called, .
+
+        Keyword arguments are forwarded directly to plugin constructors, allowing derived
+        classes to use plugins with different signatures.
+        """
+        # Make a place at the beginning for the centroid plugin to run first (because it's an OrderedDict,
+        # adding an empty element in advance means it will get run first when it's reassigned to the
+        # actual Plugin).
+        if self.config.slots.centroid != None:
+            self.plugins[self.config.slots.centroid] = None
+        # Init the plugins, sorted by execution order.  At the same time add to the schema
+        for executionOrder, name, config, PluginClass in sorted(self.config.plugins.apply()):
+            self.plugins[name] = PluginClass(config, name, metadata=self.algMetadata, **kwds)
+        # In rare circumstances (usually tests), the centroid slot not be coming from an algorithm,
+        # which means we'll have added something we don't want to the plugins map, and we should
+        # remove it.
+        if self.config.slots.centroid is not None and self.plugins[self.config.slots.centroid] is None:
+            del self.plugins[self.config.slots.centroid]
+
+    def buildSubtasks(self):
+        """Build any subtasks.
+
+        self.schema must be available when this is called
+        """
+        self.makeSubtask("applyApCorr", schema=self.schema)
```

Comment:
I don't think this provides enough convenience/code-sharing to be worthwhile, but I don't care deeply enough about it to fight it if you feel strongly about it.